### PR TITLE
Downgraded kotlinx-coroutines version

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- downgraded kotlinx-coroutines version to 1.6.4 is due to https://github.com/ktorio/ktor/pull/3609 is not released yet
- upgraded ktor version to 2.3.1

### CODE REVIEW

**Review by Cheshiriks — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -41,6 +41,26 @@ jobs:
...
+      - id: save-save-cli-version
+        name: Store save-cli version in output
+        run: |
+          sv=$(cat gradle/libs.versions.toml | grep -m1 '^save-cli =' | awk -F'[=]' '{print $2}' | tr -d '" ')
+          echo "Detected version $sv"
+          echo version=$sv >> $GITHUB_OUTPUT
+  build_cli:
+    name: Build save-cli
+    needs: [ retrieve_save_cli_version ]
+    uses: ./.github/workflows/build_save-cli_reusable.yml
```
Comment (kgevorkyan) [inline .github/workflows/build_and_test.yml:61]: do we really need the first `.`?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -41,6 +41,26 @@ jobs:
...
       gradle-cache-read-only: ${{ github.ref != 'refs/heads/master' && github.event_name != 'pull_request' }}
+  retrieve_save_cli_version:
+    name: Retrieve save-cli version from save-cloud
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.save-save-cli-version.outputs.version }}
+    steps:
+      - name: checkout save-cloud
+        uses: actions/checkout@v3
+      - id: save-save-cli-version
```
Comment (kgevorkyan) [inline .github/workflows/build_and_test.yml:52]:
```suggestion
      - id: save-cli-version
```

File: save-demo-cpg/build.gradle.kts [hunk 3]
```diff
@@ -19,6 +19,12 @@ repositories {
             artifact("/[organisation].[module]_[revision].[ext]")
         }
     }
+    maven {
+        url = uri("https://s01.oss.sonatype.org/content/repositories/snapshots/")
```
Comment (kgevorkyan) [inline save-demo-cpg/build.gradle.kts:23]: its not clear for me
we didn't used snapshot version prevoisly?

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -41,6 +41,26 @@ jobs:
     with:
       module: save-demo-agent
       gradle-cache-read-only: ${{ github.ref != 'refs/heads/master' && github.event_name != 'pull_request' }}
+  retrieve_save_cli_version:
```
Comment (kgevorkyan) [inline .github/workflows/build_and_test.yml:44]: do we removed all previous logic, corresponding for save-cli snapshots?

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -2,11 +2,11 @@
 kotlin = "1.8.22"
 jetbrains-annotations = "24.0.1"
 save-cli = "0.4.0-SNAPSHOT"
-ktor = "2.3.0"
+ktor = "2.3.1"
 okio = "3.3.0"
 serialization = "1.5.1"
 kotlinx-datetime = "0.4.0"
-kotlinx-coroutines = "1.7.1"
+kotlinx-coroutines = "1.6.4"
```
Comment (sanyavertolet) [inline gradle/libs.versions.toml:9]: And for slow ones - why do we. downgrade it?

**Review by sanyavertolet — APPROVED** [review]

File: gradle/libs.versions.toml (hunk 5 above)
Comment (sanyavertolet) [reply gradle/libs.versions.toml:9]: oh, got it. Main PR is merged, but not released

File: save-demo-cpg/build.gradle.kts (hunk 3 above)
Comment (nulls) [reply save-demo-cpg/build.gradle.kts:23]: we used, but removed when migrated to release version. also `save-demo-cpg` has overrides for repositories. that why we need to duplicate configuration here

File: .github/workflows/build_and_test.yml (hunk 4 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml:44]: yes, we refactored it during splitting save-cloud build: we had problem with long running jib on GitHub and split them to small jobs. We still have logic for save-cli in building images, but removed in regular build

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml:61]: taken from [documentation](https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow): 
```
You reference reusable workflow files using one of the following syntaxes:

- {owner}/{repo}/.github/workflows/{filename}@{ref} for reusable workflows in public and private repositories.
- ./.github/workflows/{filename} for reusable workflows in the same repository
```

But, I didn't test without leading `.`

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml:52]: done

File: .github/workflows/build_and_test_reusable.yml [hunk 6]
```diff
@@ -42,6 +42,7 @@ jobs:
         uses: gradle/gradle-build-action@v2
         env:
           SKIP_SAVE_AGENT_DEPENDENCY: true
+          SKIP_SAVE_CLI_DEPENDENCY: true
```
Comment (nulls) [inline .github/workflows/build_and_test_reusable.yml:45]: actually, we don't need to have save-cli kexe to build and run tests

**Pull request merged**